hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerRpcConnection.java

```diff
@@ -405,6 +410,16 @@ private CodedInputStream createCis(ByteBuff buf) {
   // Reads the connection header following version
   private void processConnectionHeader(ByteBuff buf) throws IOException {
     this.connectionHeader = ConnectionHeader.parseFrom(createCis(buf));
+    if (connectionHeader.getAttributeList().isEmpty()) {
```

Comment:
can you add a comment here that we want to copy the attributes prior to releasing the buffer so that they don't get corrupted down the line when the buffer's underlying memory is replaced for some other call?